lib/Dialect/FIRRTL/FIRRTLUtils.cpp

```diff
@@ -706,52 +707,48 @@ void circt::firrtl::walkGroundTypes(
   recurse(recurse, type);
 }
 
-/// Returns an operation's `inner_sym`, adding one if necessary.
-StringAttr circt::firrtl::getOrAddInnerSym(const hw::InnerSymTarget &target,
-                                           GetNamespaceCallback getNamespace) {
-
-  // Return InnerSymAttr with sym on specified fieldID.
-  auto getOrAdd = [&](auto mod, hw::InnerSymAttr attr,
-                      auto fieldID) -> std::pair<hw::InnerSymAttr, StringAttr> {
-    assert(mod);
-    auto *context = mod.getContext();
-
-    SmallVector<hw::InnerSymPropertiesAttr> props;
-    if (attr) {
-      // If already present, return it.
-      if (auto sym = attr.getSymIfExists(fieldID))
-        return {attr, sym};
-      llvm::append_range(props, attr.getProps());
-    }
+// Return InnerSymAttr with sym on specified fieldID.
+std::pair<hw::InnerSymAttr, StringAttr> circt::firrtl::getOrAddInnerSym(
+    MLIRContext *context, hw::InnerSymAttr attr, uint64_t fieldID,
```

Comment:
Nit: Is this context needed, can it be grabbed from the `hw::InnerSymAttr`, or might that be nullptr or something? Not a big deal :+1: .

Reply:
Yeah, InnerSymAttr can be `nullptr` here.  I think if we ever get to a point where we use an "empty" InnerSymAttr instead of a null one, this context can be removed.  But for now, the overwhelming majority of use cases, InnerSymAttrs can be null.